src/Vogen/Generators/Conversions/GenerateOrleansConversions.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Vogen.Generators.Conversions;
+
+public class GenerateOrleansConversions : IGenerateConversion, IGenerateAssemblyAttributes
+{
+    public string GenerateAnyAttributes(TypeDeclarationSyntax tds, VoWorkItem item) => string.Empty;
+
+    public string GenerateAnyBody(TypeDeclarationSyntax tds, VoWorkItem item)
+    {
+        if (!item.Config.Conversions.HasFlag(Vogen.Conversions.Orleans))
+        {
+            return string.Empty;
+        }
+
+        return $$"""
+                 internal sealed class OrleansSerializer : global::Orleans.Serialization.Codecs.IFieldCodec<{{item.VoTypeName}}>
+                 {
+                     public void WriteField<TBufferWriter>(ref global::Orleans.Serialization.Buffers.Writer<TBufferWriter> writer, uint fieldIdDelta, System.Type expectedType, {{item.VoTypeName}} value)
+                         where TBufferWriter : global::System.Buffers.IBufferWriter<byte>
+                     {
+                         var baseCodec = writer.Session.CodecProvider.GetCodec<{{item.UnderlyingTypeFullName}}>();
+                         baseCodec.WriteField(ref writer, fieldIdDelta, typeof({{item.UnderlyingTypeFullName}}), value.Value);
+                     }
+                 
+                     public {{item.VoTypeName}} ReadValue<TInput>(ref global::Orleans.Serialization.Buffers.Reader<TInput> reader, global::Orleans.Serialization.WireProtocol.Field field)
+                     {
+                         var baseCodec = reader.Session.CodecProvider.GetCodec<{{item.UnderlyingTypeFullName}}>();
+                         var baseValue = baseCodec.ReadValue(ref reader, field);
+                         return {{item.VoTypeName}}.From(baseValue);
```

Comment:
`From` validates the value being deserialized. Some deserializers want this, some don't. There's a `DeserializationStrictness` parameter in config that specifies how strict we are. If we call the private method `__Deserialize` and pass in the value, then that determines what should be done.

Reply:
Since Orleans serializers can end up in data stores, I believe it is beneficial that we indeed do not perform validation, just in case the consumers change their validation logic